--- a/vendor/nim-codex/library/libcodex.nim
+++ b/vendor/nim-codex/library/libcodex.nim
@@ -13,6 +13,9 @@ import
   chronicles,
  chronicles/tail

+when defined(android):
+  import android_terminal_fix
+
 export
   codex,
   codex/contracts,
@@ -20,6 +23,6 @@ export
   codex/sales,
   codex/discovery,
   codex/eth,
-  codex/utils/sequtils
+  codex/utils/sequtils

-proc initializeLibrary() {.exported.} =
+proc initializeLibrary() {.exported.} =
   discard
