--- a/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/barriers_posix.nim
+++ b/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/barriers_posix.nim
@@ -13,7 +13,7 @@ when not compileOption("threads"):
 # Types
 # -------------------------------------------------------
 
-when defined(osx):
+when defined(osx) or defined(android):
   import ./barriers_macos
   export PthreadBarrierAttr, PthreadBarrier, Errno, PTHREAD_BARRIER_SERIAL_THREAD
 else:
@@ -31,7 +31,7 @@ else:
 
 # Pthread
 # -------------------------------------------------------
-when defined(osx):
+when defined(osx) or defined(android):
   export pthread_barrier_init, pthread_barrier_wait, pthread_barrier_destroy
 else:
   # TODO careful, this function mutates `barrier` without it being `var` which
