--- a/vendor/nim-circom-compat/circomcompat.nim
+++ b/vendor/nim-circom-compat/circomcompat.nim
@@ -7,17 +7,17 @@ import
   stew/byteutils

when defined(nimHasUsed):
-  static:
-    used(circomcompat, circomcompat_runtime)
+  static:
+    used(circomcompat, circomcompat_runtime)

when defined(windows):
-  const libName = "circom_compat_ffi.dll"
+  const libName = "circom_compat_ffi.dll"
elif defined(macosx):
-  const libName = "libcircom_compat_ffi.dylib"
+  const libName = "libcircom_compat_ffi.dylib"
elif defined(android):
-  const libName = "libcircom_compat_ffi.so"
+  const libName = "libcircom_compat_ffi.so"
else:
-  const libName = "libcircom_compat_ffi.so"
+  const libName = "libcircom_compat_ffi.so"

{.passL: "-L" & currentSourcePath.parentDir / "lib".}
{.passL: "-l" & libName.}