--- a/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh	2025-12-04 21:32:53.100386993 -0500
+++ b/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh	2025-12-04 21:32:59.597357281 -0500
@@ -55,29 +55,33 @@
 	pushd "${NIM_DIR}" >/dev/null
 	if [[ -n "${NIM_COMMIT}" ]]; then
 		# support old Git versions, like the one from Ubuntu-18.04
-		git restore . 2>/dev/null || git reset --hard
-		if ! git checkout -q "${NIM_COMMIT}" 2>/dev/null; then
-			# Pay the price for a non-default NIM_COMMIT here, by fetching everything.
-			# The "upstream" remote (pointing at the same location as the "origin")
-			# is kept for backward compatibility.
-			if ! git remote | grep -q "^upstream$"; then
-				git remote add upstream https://github.com/nim-lang/Nim
+		if [[ -z "${CODEX_SKIP_GIT_RESET}" ]]; then
+			git restore . 2>/dev/null || git reset --hard
+			if ! git checkout -q "${NIM_COMMIT}" 2>/dev/null; then
+				# Pay the price for a non-default NIM_COMMIT here, by fetching everything.
+				# The "upstream" remote (pointing at the same location as the "origin")
+				# is kept for backward compatibility.
+				if ! git remote | grep -q "^upstream$"; then
+					git remote add upstream https://github.com/nim-lang/Nim
+				fi
+				# If the user has specified a custom repo, add it here as a remote as well.
+				if [[ -n "${NIM_COMMIT_REPO}" ]]; then
+					git remote remove extra 2>/dev/null || true
+					git remote add extra "${NIM_COMMIT_REPO}"
+				fi
+				git fetch --all --tags --quiet
+				git checkout -q "${NIM_COMMIT}" ||
+				{ echo "Error: wrong NIM_COMMIT specified:'${NIM_COMMIT}'"; exit 1; }
 			fi
-			# If the user has specified a custom repo, add it here as a remote as well.
-			if [[ -n "${NIM_COMMIT_REPO}" ]]; then
-				git remote remove extra 2>/dev/null || true
-				git remote add extra "${NIM_COMMIT_REPO}"
-			fi
-			git fetch --all --tags --quiet
-			git checkout -q "${NIM_COMMIT}" ||
-			  { echo "Error: wrong NIM_COMMIT specified:'${NIM_COMMIT}'"; exit 1; }
+			# In case the local branch diverged and a fast-forward merge is not possible.
+			git fetch || true
+			git reset -q --hard origin/"${NIM_COMMIT}" 2>/dev/null || true
+			# In case NIM_COMMIT is a local branch that's behind the remote one it's tracking.
+			git pull -q 2>/dev/null || true
+			git checkout -q "${NIM_COMMIT}"
+		else
+			echo "Skipping git reset/checkout operations to preserve patches"
 		fi
-		# In case the local branch diverged and a fast-forward merge is not possible.
-		git fetch || true
-		git reset -q --hard origin/"${NIM_COMMIT}" 2>/dev/null || true
-		# In case NIM_COMMIT is a local branch that's behind the remote one it's tracking.
-		git pull -q 2>/dev/null || true
-		git checkout -q "${NIM_COMMIT}"
 		# We can't use "rev-parse" here, because it would return the tag object's
 		# hash instead of the commit hash, when NIM_COMMIT is a tag.
 		NIM_COMMIT_HASH="$(git rev-list -n 1 "${NIM_COMMIT}")"
