--- a/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh
+++ b/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh
@@ -45,7 +45,7 @@
   # Update csources if needed
   if [ ! -f "$csourcesDir"/build.sh ]; then
     echo "Updating csources..."
-    git pull -q
+    echo "Skipping git pull - patches are applied"
   fi
 
   # Build csources
--- a/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh
+++ b/vendor/nim-codex/vendor/nimbus-build-system/scripts/build_nim.sh
@@ -53,7 +53,7 @@
   # Update Nimble packages if needed
   if [ ! -f "$nimbleDir"/nimblepkg.json ]; then
     echo "Updating nimble packages..."
-    git pull -q
+    echo "Skipping git pull - patches are applied"
   fi
 
   # Build nimble