--- a/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/barriers_posix.nim
+++ b/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/barriers_posix.nim
@@ -13,7 +13,7 @@
 # Types
 # -------------------------------------------------------
 
-when defined(osx):
+when defined(osx) or defined(android):
   import ./barriers_macos
   export PthreadBarrierAttr, PthreadBarrier, Errno, PTHREAD_BARRIER_SERIAL_THREAD
 else:
