--- a/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/allocs.nim
+++ b/vendor/nim-codex/vendor/nim-taskpools/taskpools/primitives/allocs.nim
@@ -105,7 +105,7 @@
   proc aligned_alloc_windows(size, alignment: csize_t): pointer {.sideEffect,importc:"_aligned_malloc", header:"<malloc.h>".}
     # Beware of the arg order!
   proc tp_freeAligned*[T](p: ptr T){.sideEffect,importc:"_aligned_free", header:"<malloc.h>".}
-elif defined(osx):
+elif defined(osx) or defined(android):
   proc posix_memalign(mem: var pointer, alignment, size: csize_t){.sideEffect,importc, header:"<stdlib.h>".}
   proc aligned_alloc(alignment, size: csize_t): pointer {.inline.} =
     posix_memalign(result, alignment, size)
