--- a/vendor/nim-codex/vendor/nim-nat-traversal/vendor/libnatpmp-upstream/Makefile
+++ b/vendor/nim-codex/vendor/nim-nat-traversal/vendor/libnatpmp-upstream/Makefile
@@ -5,7 +5,7 @@
 # http://miniupnp.free.fr/libnatpmp.html
 
 OS = $(shell $(CC) -dumpmachine)
-CC ?= gcc
+CC ?= $(shell if [ "$(ANDROID_NDK_HOME)" != "" ] && [ "$(TARGET_ARCH)" = "arm64" ]; then echo "$(ANDROID_NDK_HOME)/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android21-clang"; else echo "gcc"; fi)
 ARCH = $(OS)
 VERSION = $(shell cat VERSION)
 INSTALL ?= $(shell which install)
@@ -32,6 +32,10 @@
 CFLAGS += -Wall
 CFLAGS += -Wextra
 CFLAGS += -DENABLE_STRNATPMPERR
+ifeq ($(TARGET_ARCH),arm64)
+CFLAGS += -DANDROID
+CFLAGS += -fPIC
+endif
 
 LIBOBJS = natpmp.o getgateway.o
 
 OBJS = $(LIBOBJS) testgetgateway.o natpmpc.o natpmp-jni.o
