--- a/vendor/nim-codex/vendor/nimbus-build-system/vendor/Nim/lib/system/sysatomics.nim.orig	2025-11-25 20:56:53.289880549 -0500
+++ b/vendor/nim-codex/vendor/nimbus-build-system/vendor/Nim/lib/system/sysatomics.nim	2025-11-25 20:57:22.134277064 -0500
@@ -357,6 +357,10 @@
 elif (defined(x86) or defined(amd64)) and (someGcc or defined(bcc)):
   proc cpuRelax* {.inline.} =
     {.emit: """asm volatile("pause" ::: "memory");""".}
+elif defined(arm64) and (someGcc or defined(bcc)):
+  proc cpuRelax* {.inline.} =
+    # Use yield instruction for ARM64 instead of pause
+    {.emit: """asm volatile("yield" ::: "memory");""".}
 elif someGcc or defined(tcc):
   proc cpuRelax* {.inline.} =
     {.emit: """asm volatile("" ::: "memory");""".}
