--- a/vendor/constantine/constantine/platforms/isa_x86/cpudetect_x86.nim
+++ b/vendor/constantine/constantine/platforms/isa_x86/cpudetect_x86.nim
@@ -17,6 +17,7 @@ proc cpuid(eax: uint32, ecx = 0'u32): CpuIdRegs =
   ## Query the CPU
   ##
   ## CPUID is a very slow operation, 27-70 cycles, ~120 latency
+when not defined(android) and not defined(arm64) and not defined(arm):
   ##   - https://uops.info/table.html
   ##   - https://www.agner.org/optimize/instruction_tables.pdf
   ##
@@ -38,6 +39,7 @@ proc cpuid(eax: uint32, ecx = 0'u32): CpuIdRegs =
        :"a"(`eax`), "c"(`ecx`)"""
 
 # CPU Name
+# ----------------------------------------------------------------------
 
 proc cpuName_x86*(): string =
+when not defined(android) and not defined(arm64) and not defined(arm):
   let leaves = cast[array[48, char]]([
     cpuid(0x80000002'u32),
     cpuid(0x80000003'u32),
@@ -120,6 +122,7 @@ proc detectCpuFeaturesX86() {.loadTime.} =
   # see: Intel® Architecture Instruction Set Extensions and Future Features Programming Reference
   #      2023-09: https://cdrdv2-public.intel.com/790021/architecture-instruction-set-extensions-programming-reference.pdf
 
+when not defined(android) and not defined(arm64) and not defined(arm):
   # leaf 1, ecx
   hasSse3Impl             = leaf1.ecx.test(0)
   hasSsse3Impl            = leaf1.ecx.test(9)
@@ -183,6 +186,7 @@ proc detectCpuFeaturesX86() {.loadTime.} =
 
 # 1999 - Pentium 3, 2001 - Athlon XP
 # ------------------------------------------
+when not defined(android) and not defined(arm64) and not defined(arm):
 proc hasSse*(): bool {.inline.} =
   return hasSseImpl
 
@@ -303,3 +307,4 @@ proc hasAvx512bitalg*(): bool {.inline.} =
   ## AVX512 Bit ALgorithm
   return hasAvx512bitalgImpl
+when not defined(android) and not defined(arm64) and not defined(arm):