--- a/vendor/nim-codex/codex/blockexchange/engine/engine.nim
+++ b/vendor/nim-codex/codex/blockexchange/engine/engine.nim
@@ -369,7 +369,7 @@
           else:
             0.milliseconds
 
         let retryDelay =
-          max(secs(rand(self.pendingBlocks.retryInterval.secs)), nextDiscovery)
+          max(secs(rand(int(self.pendingBlocks.retryInterval.secs)).int), nextDiscovery)
 
         # We now wait for a bit and then retry. If the handle gets completed in the
         # meantime (cause the presence handler might have requested the block and
         # got it from another peer) the handle will be removed from the pending
         # set and the retry will be cancelled.
